wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java

```diff
@@ -474,6 +552,200 @@ public void test_readerLoadBalancing_autocommitFalse() throws SQLException {
+  public void test_readerLoadBalancing_remainingStateTransitions(final Properties props) throws SQLException {
+
+    ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.set(props, "true");
+    final String url = getUrl();
+    LOGGER.finest("Connecting to url " + url);
+    try (final Connection conn = DriverManager.getConnection(url, this.defaultProps)) {
+
+      final String writerConnectionId = queryInstanceId(conn);
+      LOGGER.finest("writerConnectionId: " + writerConnectionId);
+
+      conn.setReadOnly(true);
+      conn.setReadOnly(false);
+      conn.setReadOnly(true);
+      conn.setAutoCommit(false);
+      conn.setAutoCommit(true);
+      Statement stmt = conn.createStatement();
+      stmt.execute("commit");
+      stmt = conn.createStatement();
+      stmt.execute("commit");
+      stmt = conn.createStatement();
+      stmt.execute("begin");
+      stmt.execute("commit");
+      conn.setAutoCommit(false);
+      conn.commit();
+      conn.commit();
+      stmt = conn.createStatement();
+      stmt.execute("begin");
+      stmt.execute("SELECT 1");
+      conn.commit();
+      conn.setReadOnly(false);
+      conn.setReadOnly(true);
+      conn.setReadOnly(false);
+      conn.setAutoCommit(true);
+      conn.setReadOnly(false);
+      stmt = conn.createStatement();
+      stmt.execute("commit");
+      conn.setReadOnly(false);
+      conn.setReadOnly(true);
+      conn.setAutoCommit(false);
+      conn.commit();
+      conn.setAutoCommit(true);
+    }
+  }
+
+  @TestTemplate
+  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  public void test_readerLoadBalancing_lostConnectivity(final Properties props) throws SQLException {
+
+    ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.set(props, "true");
+    final String url = getUrl();
+    LOGGER.finest("Connecting to url " + url);
+    try (final Connection conn = DriverManager.getConnection(url, this.defaultProps)) {
+
+      final String writerConnectionId = queryInstanceId(conn);
+      LOGGER.finest("writerConnectionId: " + writerConnectionId);
+
+      conn.setReadOnly(true);
+      final Statement stmt1 = conn.createStatement();
+      // autocommit on transaction (autocommit implicitly disabled)
+      stmt1.execute("BEGIN");
+      final String readerId = queryInstanceId(conn);
+
+      ProxyHelper.disableConnectivity(
+          TestEnvironment.getCurrent()
+              .getInfo()
+              .getDatabaseInfo()
+              .getInstances()
+              .get(1)
+              .getInstanceId());
+
+      final SQLException e = assertThrows(SQLException.class, () -> queryInstanceId(conn));
+      ProxyHelper.enableConnectivity(
+          TestEnvironment.getCurrent()
+              .getInfo()
+              .getDatabaseInfo()
+              .getInstances()
+              .get(1)
+              .getInstanceId());
+
+      if (pluginChainIncludesFailoverPlugin(props)) {
```

Comment:
Connection configuration should be known at this moment. Why to check properties?